pkg/resource/deploy/source_eval.go

```diff
@@ -1422,6 +1433,22 @@ func (rm *resmon) RegisterResource(ctx context.Context,
 	}
 
 	additionalSecretOutputs := req.GetAdditionalSecretOutputs()
+	if rm.loader != nil {
```

Comment:
Should loader always be non-nil by this point?

Reply:
Yeah I think so, since it is being passed from `evalSource` and that requires a non-loader to initialize. I'll revisit once I fix the integration tests